server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueViewConfiguration.java

```diff
@@ -0,0 +1,95 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.view.cassandra;
+
+import java.time.Duration;
+
+import com.google.common.base.Preconditions;
+
+public class CassandraMailQueueViewConfiguration {
+    interface Builder {
+        @FunctionalInterface
+        interface RequireBucketCount {
+            RequireUpdateBrowseStartPace bucketCount(int bucketCount);
+        }
+
+        @FunctionalInterface
+        interface RequireUpdateBrowseStartPace {
+            RequireSliceWindow updateBrowseStartPace(int updateBrowseStartPace);
+        }
+
+        @FunctionalInterface
+        interface RequireSliceWindow {
+            LastStage sliceWindow(Duration sliceWindow);
+        }
+
+        class LastStage {
```

Comment:
usually, we use the Builder object for that

Reply:
Ok, I'm actually the one proposing this...

I really tend to like the cascading lambdas for enforcing compulsory value in a builder.

But because we need to define many interfaces, I find it hard to understand what is in the builder and what is not.... I really find it useful to get an API packing it together. At first sight I can see what's part of the builder and what is not.

Then for naming... Is the builder the succession of stages leading to the object creation, **or** is it just the final generation operation? 

The question was pointless before cascading lambda use, as we had a single object.

But I would propose a 50-50 naming. How about `s/LastStage/TerminatingBuilder/` and keeping all of these freaks in the Builder interface?

Reply:
I see your point : we have actually two builders. One builder that is the cascading lambda, and the builder for optional parameters. In fact we have many builders, one for each stage (all interfaces are builders).
In any way,  the outer interface is not a builder, we have to find a name for it, say BuilderStages.
We could rename LastStage to ReadyToBuild ? (that would fit with Require* then ReadyToBuild because requirements are fulfilled)